File: 7-rounding/task.erl
-module(task).

-export([round/2]).

round(Count, Values) ->
  round(Count, Values, []).

round(_, [], Result) ->
  lists:reverse(Result);
round(Count, [{Value1, Value2}|ValuesTail], Result) ->
  round(Count - 1, ValuesTail, [erlang:round(Value1 / Value2)|Result]).

File: 7-rounding/task.ex
defmodule Task do

  def round(count, values) do
    round(count, values, [])
  end

  defp round(_, [], result) do
    Enum.reverse(result)
  end

  defp round(count, [{value1, value2}|tail_values], result) do
    round(count - 1, tail_values, [Kernel.round(value1 / value2)|result])
  end

end
